cpp/src/stream_compaction/distinct_count.cu

```diff
@@ -0,0 +1,196 @@
+/*
```

Comment:
This is not new code but was just moved from the `drop_duplicates.cu` source file here to its own source file.